test/resource_owner_password_grant_test.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+@TestOn("vm")
+library resource_owner_password_grant_test;
+
+import 'dart:convert';
+import 'dart:async';
+import 'package:http/http.dart' as http;
+import 'package:crypto/crypto.dart';
+
+import 'package:test/test.dart';
+import 'package:oauth2/oauth2.dart' as oauth2;
+import 'utils.dart';
+
+final String SUCCESS = JSON.encode({
+  "access_token": "2YotnFZFEjr1zCsicMWpAA",
+  "token_type": "bearer",
+  "expires_in": 3600,
+  "refresh_token": "tGzv3JOkF0XG5Qx2TlKWIA",
+});
+
+var auth = 'Basic ${CryptoUtils.bytesToBase64(UTF8.encode('client:secret'))}';
+var authEndpoint = Uri.parse('https://example.com');
+var expectClient = new ExpectClient();
+
+void main() {
+  group('basic', () {
+    test('builds correct request with client when using basic auth for client',
+        () async {
+      expectClient.expectRequest((request) {
+        expect(auth, equals(request.headers['authorization']));
+        expect(request.url.queryParameters['grant_type'], equals('password'));
+        expect(request.url.queryParameters['username'], equals('username'));
+        expect(request.url.queryParameters['password'], equals('userpass'));
+        return new Future.value(new http.Response(SUCCESS, 200,
```

Comment:
Rather than using `new Future.value` here, just make the request-handling block `async`.